File: Notes/app/src/main/java/com/serhii/apps/notes/ui/utils/TextChecker.java
package com.serhii.apps.notes.ui.utils;

import android.text.Editable;
import android.text.TextUtils;
import android.text.TextWatcher;
import android.widget.Button;
import android.widget.EditText;

/**
 * Helper class for enabling button if edit text has entered text
 */
public class TextChecker implements TextWatcher {

    private final EditText controlField;
    private final Button controlButton;

    public TextChecker(EditText controlField, Button controlButton) {
        this.controlField = controlField;
        this.controlButton = controlButton;
    }

    @Override
    public void beforeTextChanged(CharSequence s, int start, int count, int after) {

    }

    @Override
    public void onTextChanged(CharSequence s, int start, int before, int count) {

    }

    @Override
    public void afterTextChanged(Editable s) {
        boolean enable = !TextUtils.isEmpty(controlField.getText());
        controlButton.setEnabled(enable);
    }

}

File: Notes/app/src/main/java/com/serhii/apps/notes/ui/utils/TextChecker.kt
/*
 * Copyright 2022. Happy coding ! :)
 * Author: Serhii Butryk
 */
package com.serhii.apps.notes.ui.utils

import android.widget.EditText
import android.text.TextWatcher
import android.text.Editable
import android.text.TextUtils
import android.widget.Button

/**
 * Helper class for enabling button if text has changed
 */
class TextChecker(private val controlField: EditText, private val controlButton: Button) : TextWatcher {

    override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}

    override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}

    override fun afterTextChanged(s: Editable) {
        val enable = !TextUtils.isEmpty(controlField.text)
        controlButton.isEnabled = enable
    }

}